### PR TITLE
Hibernate intro

### PR DESCRIPTION
**Pull request by Ketch92**

### CODE REVIEW

File: src/main/java/core/Main.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import core.service.MovieService;
+
+public class Main {
+    private static Injector injector = Injector.getInstance(Main.class.getPackageName());
+    
+    public static void main(String[] args) {
+        MovieService movieService = (MovieService) injector.getInstance(MovieService.class);
+        Movie movie = new Movie();
+        movie.setTitle("Fast and Furious");
+
```
Comment (gigbat) [inline src/main/java/core/Main.java]:
```suggestion
        movie.setTitle("Fast and Furious");
        movieService.add(movie);
```

File: src/main/java/core/dao/MovieDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+    
+    @Override
+    public List<Movie> getAll() {
+        Transaction transaction = null;
```
Comment (gigbat) [inline src/main/java/core/dao/MovieDaoImpl.java]: unused operation here

**Review by gigbat — CHANGES_REQUESTED** [review]

File: src/main/java/core/Main.java (hunk 1 above)
Comment (Ketch92) [reply src/main/java/core/Main.java]: LOL

File: src/main/java/core/dao/MovieDaoImpl.java (hunk 2 above)
Comment (Ketch92) [reply src/main/java/core/dao/MovieDaoImpl.java]: True

**Review by gigbat — APPROVED** [review]

File: src/main/java/core/Main.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import core.model.Movie;
+import core.service.MovieService;
+
+public class Main {
+    private static Injector injector = Injector.getInstance(Main.class.getPackageName());
+    
+    public static void main(String[] args) {
+        MovieService movieService = (MovieService) injector.getInstance(MovieService.class);
+        Movie movie = new Movie();
+        movie.setTitle("Fast and Furious");
```
Comment (Sofasmile) [inline src/main/java/core/Main.java:13]: let's test add() method here

File: src/main/java/core/service/MovieServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import core.lib.Inject;
+import core.lib.Service;
+import core.model.Movie;
+import java.util.List;
+
+@Service
+public class MovieServiceImpl implements MovieService {
+    
+    @Inject
+    private MovieDao dao;
```
Comment (Sofasmile) [inline src/main/java/core/service/MovieServiceImpl.java]:
```suggestion
    private MovieDao movieDao;
```

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/core/Main.java (hunk 3 above)
Comment (Ketch92) [reply src/main/java/core/Main.java:13]: Somehow I removed it, Thanks

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**